alacritty_terminal/src/term/mod.rs

```diff
@@ -280,11 +280,6 @@ impl<T> Term<T> {
         self.grid.scroll_display(scroll);
         self.event_proxy.send_event(Event::MouseCursorDirty);
 
-        // Clamp vi mode cursor to the viewport.
-        let viewport_start = -(self.grid.display_offset() as i32);
-        let viewport_end = viewport_start + self.bottommost_line().0;
-        let vi_cursor_line = &mut self.vi_mode_cursor.point.line.0;
-        *vi_cursor_line = min(viewport_end, max(viewport_start, *vi_cursor_line));
```

Comment:
Why did you remove this? I believe this is still necessary in some scenarios.

Reply:
Can you show me an example? We should add it as a test if it exists.

Reply:
Scrolling with the mouse. This is definitely still required.